index.html

```diff
@@ -495,6 +495,462 @@ <h2>
+          </li>
+          <li>[=map/for each=] |mime_type_string:string| → |extensions| of
+          |item|["accept"]
+            <ol>
+              <li>If |extensions| is not a [=list=], continue.
+              </li>
+              <li>If |extensions| is [=list/empty=], continue.
+              </li>
+              <li>If |extensions| [=list/contains=] items that are not
+              [=string=]s, continue.
+              </li>
+              <li>If |extensions| [=list/contains=] strings that do not begin
+              with the character `.`, continue.
+              </li>
+              <li>Let |mime_type_parsed:mime type| be the result of running the
+              steps of [=parse a mime type=] on |mime_type_string|.
+              </li>
+              <li>If |mime_type_parsed:mime type| is failure, continue.
+              </li>
+              <li>If |mime_type_parsed/type| is not listed as a top-level type
+              in in [[IANA-MEDIA-TYPES]], continue.
+              </li>
+              <li>Set |accept|[|mime_type_string|] to |extensions|.
+              </li>
+            </ol>
+          </li>
+          <li>If |accept:ordered map| is empty, return failure.
+          </li>
+          <li>Let |file_handler:ordered map| be |ordered map| «[ "action" →
+          |url|, "name" → |item|["name"], "launch_type" → |launch_type|,
+          "launch_type" "accept" → |accept| ]».
+          </li>
+          <li>
+            <a data-cite="appmanifest#dfn-process-image-resources">Process
+            image resources</a> passing |item|["icons"], |file_handler|,
+            |manifest URL|, and "icons".
+          </li>
+          <li>Return |file_handler|.
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h2>
+          Execute a file handler launch
+        </h2>
+        <p>
+          The steps to <dfn>execute a file handler launch</dfn> are given by
+          the following algorithm. The algorithm takes [=list=] |files:list|
+          and a [=ordered map=] |manifest:ordered_map| which holds results from
+          [=processing a manifest=].
+        </p>
+        <ol class="algorithm">
+          <li>Let |file_handlers:list| be |manifest|["file_handlers"].
+          </li>
+          <li>If |file_handlers:list| is null, return.
+          </li>
+          <li>Let |launches:ordered map| be an [=ordered map=].
+          </li>
+          <li>[=list/for each=] |filename:string| of |files|
+            <ol>
+              <li>[=list/for each=] |file_handler:ordered_map| of
+              |file_handlers:list|
+                <ol>
+                  <li>[=map/for each=] |mime_type_string:string| →
+                  |extensions:list| of |file_handler|["accept"]
+                    <ol>
+                      <li>[=list/for each=] |extension:string| of |extensions|
+                        <ol>
+                          <li>If |filename| does not end in |extension|,
```

Comment:
Maybe we should do trimming of filenames? We used to do trimming in web app manifest, so might be good to check what we do there.

Reply:
Do you mean whitespace trimming? The file names are provided by the OS as command line arguments. AFAIK there's no rule saying a file name can't start with a space.